docs/docs/modules/model_io/chat/function_calling.mdx

```diff
@@ -72,78 +121,178 @@ import ChatModelTabs from "@theme/ChatModelTabs";
+```
 
 ```python
-llm_with_tools = llm.bind_tools([Multiply], tool_choice="Multiply")
-llm_with_tools.invoke("what's 3 * 12")
+print(type(gathered.tool_call_chunks[0]["args"]))
 ```
 
 ```text
-AIMessage(content='', additional_kwargs={'tool_calls': [{'index': 0, 'id': 'call_qIP2bJugb67LGvc6Zhwkvfqc', 'type': 'function', 'function': {'name': 'Multiply', 'arguments': '{"a": 3, "b": 12}'}}]})
+<class 'str'>
 ```
 
-If we wanted to force that a tool is used (and that it is used only
-once), we can set the `tool_choice` argument to the name of the tool:
+And below we accumulate tool calls to demonstrate partial parsing:
 
 ```python
-llm_with_multiply = llm.bind_tools([Multiply], tool_choice="Multiply")
-llm_with_multiply.invoke(
-    "make up some numbers if you really want but I'm not forcing you"
-)
+first = True
+async for chunk in llm_with_tools.astream(query):
+    if first:
+        gathered = chunk
+        first = False
+    else:
+        gathered = gathered + chunk
+
+    print(gathered.tool_calls)
 ```
 
 ```text
-AIMessage(content='', additional_kwargs={'tool_calls': [{'id': 'call_f3DApOzb60iYjTfOhVFhDRMI', 'function': {'arguments': '{"a":5,"b":10}', 'name': 'Multiply'}, 'type': 'function'}]})
+[]
+[]
+[{'name': 'Multiply', 'args': {}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}]
+[{'name': 'Multiply', 'args': {'a': 3}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 1}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 12}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 12}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 12}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}, {'name': 'Add', 'args': {}, 'id': 'call_zPAyMWr8hN1q083GWGX2dSiB'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 12}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}, {'name': 'Add', 'args': {'a': 11}, 'id': 'call_zPAyMWr8hN1q083GWGX2dSiB'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 12}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}, {'name': 'Add', 'args': {'a': 11}, 'id': 'call_zPAyMWr8hN1q083GWGX2dSiB'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 12}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}, {'name': 'Add', 'args': {'a': 11, 'b': 49}, 'id': 'call_zPAyMWr8hN1q083GWGX2dSiB'}]
+[{'name': 'Multiply', 'args': {'a': 3, 'b': 12}, 'id': 'call_z3B4o82SQDY5NCnmrXIcVQo4'}, {'name': 'Add', 'args': {'a': 11, 'b': 49}, 'id': 'call_zPAyMWr8hN1q083GWGX2dSiB'}]
 ```
 
-For more see the [ChatOpenAI API
-reference](https://api.python.langchain.com/en/latest/chat_models/langchain_openai.chat_models.base.ChatOpenAI.html#langchain_openai.chat_models.base.ChatOpenAI.bind_tools).
+```python
+print(type(gathered.tool_calls[0]["args"]))
+```
+
+```text
+<class 'dict'>
+```
 
 ## Defining functions schemas
```

Comment:
shoudl we delete the rest of this doc, user shouldn't have to think about function schemas anymore right?

Reply:
@ccurme  added it, I think dto underscore why the standard interface is nice. Don't feel super strongly though